packages/ui/src/utils/get-field-groups.ts

```diff
@@ -0,0 +1,22 @@
+import { useMemo } from 'react';
+import { getValue } from '.';
+
+export const useFieldGroups = (fields: { [name: string]: unknown }) => {
+  const propertiesArray = useMemo(() => {
```

Comment:
We shouldn't use the `useMemo` hook here, instead, just return the function to generate the object and use the `useHook` in the desired components.

Alternatively, if you still want to use a `useMemo` here, then this file needs to be turned into a custom hook, for more information, please check the `hooks` folder